cpp/src/parquet/encryption/internal_file_decryptor.cc

```diff
@@ -168,21 +172,7 @@ std::shared_ptr<Decryptor> InternalFileDecryptor::GetColumnDataDecryptor(
 std::shared_ptr<Decryptor> InternalFileDecryptor::GetColumnDecryptor(
     const std::string& column_path, const std::string& column_key_metadata,
     const std::string& aad, bool metadata) {
-  std::string column_key;
```

Comment:
This may introduce performance regression. But that's not my concern here due to keeping the cache in single-threaded mode may require more complicated logic.

Reply:
Would it make sense to at least cache `column_key` though? As mentioned in the PR description, `key_retriever()->GetKey` might call out to the user's key management service and could be fairly expensive.

Reply:
Although if double wrapping is used, it looks like keys will be cached with `kek_per_kek_id_` in `FileKeyUnwrapper` so maybe the performance impact won't be too bad.